tests/e2e/utils/DriverHelper.ts

```diff
@@ -310,99 +340,135 @@ export class DriverHelper {
+                        await this.wait(polling);
+                        continue;
+                    }
+
+                    throw err;
+                }
+            } catch(err) {
+                if (i >= attempts) {
+                    Logger.error(`DriverHelper.type - failed with exception, out of attempts - ${err}`);
+                    throw err;
+                } else {
                     continue;
                 }
-
-                throw err;
             }
         }
 
         throw new error.TimeoutError(`Exceeded maximum typing attempts, to the '${elementLocator}' element`);
     }
 
     public async typeToInvisible(elementLocator: By, text: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
-        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
         const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
 
         Logger.trace(`DriverHelper.typeToInvisible ${elementLocator} text: ${text}`);
 
         for (let i = 0; i < attempts; i++) {
-            const element: WebElement = await this.waitPresence(elementLocator, timeout);
-
             try {
-                await element.sendKeys(text);
-                return;
-            } catch (err) {
-                if (err instanceof error.StaleElementReferenceError) {
-                    await this.wait(polling);
+                const element: WebElement = await this.waitPresence(elementLocator, polling);
+
+                try {
+                    await element.sendKeys(text);
+                    return;
+                } catch (err) {
+                    if (err instanceof error.StaleElementReferenceError) {
+                        await this.wait(polling);
+                        continue;
+                    }
+
+                    throw err;
+                }
+            } catch(err) {
+                if (i >= attempts) {
+                    Logger.error(`DriverHelper.typeToInvisible - failed with exception, out of attempts - ${err}`);
+                    throw err;
+                } else {
                     continue;
                 }
-
-                throw err;
             }
         }
 
         throw new error.TimeoutError(`Exceeded maximum typing attempts, to the '${elementLocator}' element`);
     }
 
     public async clear(elementLocator: By, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
-        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
         const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
 
         Logger.trace(`DriverHelper.clear ${elementLocator}`);
 
         for (let i = 0; i < attempts; i++) {
-            const element: WebElement = await this.waitVisibility(elementLocator, timeout);
-
             try {
```

Comment:
2 levels of try/catch

Reply:
I had issues for the methods failing after times that did not match the timeout without a sensual exception message.
The waitVisibility method was also failing with ElementNotFound that was never caught and caused issue with the logic of the tests.
The second layer of try-catch is there to prevent the method failing without reaching the desired amount of attempts/timeout and if the issue persists, it throws the correct exception.